Classes/commands.py

```diff
@@ -687,7 +687,99 @@ async def remove_inactive_cadets(self, ctx, inactive_days_required):
+
+        loa_officer_ids = []
+
+        # If the entry is still good, add the officer to our exclusion list. Otherwise, delete the entry if expired.
+        for entry in loa_entries:
+            if entry[2] > datetime.utcnow().date():
+                if entry[3]: loa_officer_ids.append(entry[0])
+            else:
+                await Officer.remove_loa(self.bot, str(entry[4]))
+
+        # For everyone in the server where their role is in the role ladder,
+        # get their last activity times, or if no last activity time, use
+        # the time we started monitoring them. Exclude those we have already
+        # determined have a valid Leave of Absence
+
+        # Get a date range for our LOAs, and make some dictionaries to work in
+        max_inactive_days = bot.settings["max_inactive_days"]
+        oldest_valid = datetime.utcnow() - timedelta(days=max_inactive_days)
+        inactive_officers = []
+        role_ids = role_id_index(bot.settings)
+        officers_to_check = []
+        guild = self.bot.officer_manager.guild
+        
+        for officer in self.bot.officer_manager.all_officers:
+            if officer.id not in loa_officer_ids:
+                last_activity = await officer.get_last_activity(ctx.bot.officer_manager.all_monitored_channels)
+                last_activity = last_activity["time"]
+                try:
+                    if last_activity < oldest_valid:
+                        inactive_officers.append(officer)
+                except:
+                    await ctx.channel.send("There was a problem with the activity times. Make sure that there are officers with patrol times", delete_after=10)
+        
+        
+        role = self.bot.officer_manager.guild.get_role(
+            self.bot.settings["inactive_role"]
+        )
+        for member in inactive_officers:
+            confirm = await Confirm(f'Do you want to mark {member.mention} as LPD Inactive?').prompt(ctx)
+            if confirm:
+                await member.add_roles(role)
+                await ctx.channel.send(f'{member.mention} has been marked LPD_inactive')
+            else:
+                await ctx.channel.send(f'{member.mention} will have their inactivity reevaluated at a later date.')
+            
+        inactive_officers = []
+
+
 
+    @checks.is_admin_bot_channel()
+    @checks.is_white_shirt()
+    @commands.command()
+    # Review Leaves of Absence
+    async def show_loa(self, ctx):
+        """
+        This command displays all Leave of Absense requests currently on file.
+        """
+        loa_entries = await self.bot.officer_manager.get_loa()
+        i = 0
+        for entry in loa_entries:
+            i = i + 1
+            officer = self.bot.get_user(entry[0])
+            string = f"There are currently Leaves of Absence on file for the following Officers:"
+            string = f"{string}\n{officer.mention} from {entry[1]} to {entry[2]} for reason: {entry[3]}"
+            if len(string) > 1000:
```

Comment:
In stead of implementing this functionality each time, and having the chance of messing up, you can use the send_long function. You don't need to change this though, just something to think about in the future.